server/resource_server/controller/controller.go

```diff
@@ -239,6 +270,34 @@ func LoginUserHandler(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func LoginUserHandlerv2(w http.ResponseWriter, r *http.Request) {
+	if !validateHttpMethod(w, r.Method, http.MethodPost) {
+		return
+	}
+
+	newService := r.Context().Value("service").(interfaces.IService)
+
+	request, err := utils.DecodeJsonFromRequest[dto.LoginUserDTOv2](w, r.Body)
+	if err != nil {
+		return
+	}
+
+	serverSignatureResp, err := newService.LoginUserv2(request)
+	if err != nil {
+		utils.HandleError(w, http.StatusBadRequest, "Failed to perform login", err)
+		return
+	}
+
```

Comment:
Same here, would it make sense to encode `serverSignatureResp` into `Response` object?